daemon/src/wallet.rs

```diff
@@ -0,0 +1,50 @@
+use crate::model::{WalletError, WalletInfo};
+use anyhow::Result;
+use bdk::bitcoin::Amount;
+use bdk::blockchain::{Blockchain, NoopProgress};
+use bdk::database::BatchDatabase;
+use bdk::wallet::AddressIndex;
+use tokio::sync::watch;
+
+pub fn wallet_sync<B, D>(
+    wallet_feed_sender: &watch::Sender<WalletInfo>,
+    wallet: &bdk::Wallet<B, D>,
+    previous: WalletInfo,
+) -> Result<Option<WalletInfo>, WalletError>
+where
+    B: Blockchain,
+    D: BatchDatabase,
+{
+    wallet
+        .sync(NoopProgress, None)
+        .map_err(|_| WalletError::Sync)?;
+
+    let balance = wallet.get_balance().map_err(|_| WalletError::Balance)?;
+
+    let update = match previous {
+        WalletInfo::Update {
+            balance: prev_balance,
+            ..
+        } => balance != prev_balance.as_sat(),
+        WalletInfo::Syncing | WalletInfo::Error(_) => true,
+    };
```

Comment:
Do we need this?

Reply:
That's the logic that determines if a wallet update should be sent or not. I thought this is what we had discussed :D
If the balance did not change we don't send and update into the feed (unless we were previously in an Error or Syncing state.
I think Syncing is actually only used initially now, so we could rename it (previously I had used it more, but that did not make sense).

Reply:
removed